src/main/java/com/github/wovnio/wovnjava/HtmlConverter.java

```diff
@@ -44,20 +47,29 @@ String convert(String lang) {
     }
 
     String restore(String html) {
-        StringBuilder sb = new StringBuilder();
-        String[] list = html.split("<!--" + WOVN_MARKER_PREFIX);
-
-        sb.append(list[0]);
-        for (int i=1; i<list.length; i++) {
-            String fragment = list[i];
-            String commentSuffix = "-->";
-            int suffixOffset = fragment.indexOf(commentSuffix);
-            String indexString = fragment.substring(0, suffixOffset);
-            int index = Integer.parseInt(indexString);
-            sb.append(markers.get(index));
-            sb.append(fragment.substring(suffixOffset + commentSuffix.length()));
+        Document document = Jsoup.parse(html);
```

Comment:
The point of the html string is we don't want to re-parse the document again, ideally.

Reply:
Ok.